src/photon/photon_algorithm.c

```diff
@@ -17,26 +17,27 @@ typedef struct task_queue_entry {
   struct task_queue_entry *next;
 } task_queue_entry;
 
+/** A data structure used to track which objects are available locally and
+ *  which objects are being actively fetched. */
 typedef struct {
   /** Object id of this object. */
   object_id object_id;
-  /** Handle for the uthash table. */
-  UT_hash_handle handle;
-} available_object;
+  /** Handle for the uthash table. NOTE: This handle is used for both the
+   *  scheduling algorithm state's local_objects and fetch_requests tables.
+   *  We must reinforce the uthash invariant that the entry be in either one or
+   *  neither tables. */
```

Comment:
It's okay to use the same handle as long as the entry isn't in both tables at the same time.